File: selfstudy/js/temperatures_reverse.js
/*
task 48:
Temperatures - Reverse
You need to reverse the order of daily temperatures in Celsius
*/

function reverse_order_temperatures(temps) {
  return [...temps].reverse();
}

// Tests
const tests = [
  { input: [10, 20, 30], expected: reverse_order_temperatures([10, 20, 30]) },
  { input: [50, 60, 70], expected: reverse_order_temperatures([50, 60, 70]) },
  { input: [1, 2, 3], expected: reverse_order_temperatures([1, 2, 3]) },
  { input: [100, 200], expected: reverse_order_temperatures([100, 200]) },
  { input: [], expected: reverse_order_temperatures([]) }
];

if (process.argv.includes('mocha')) {
  const assert = require('assert');
  describe('reverse_order_temperatures', () => {
    tests.forEach((t, idx) => {
      it(`test case ${idx+1}`, () => {
        assert.deepStrictEqual(reverse_order_temperatures(t.input), t.expected);
      });
    });
  });
} else {
  tests.forEach((t, idx) => {
    console.log(`Test ${idx+1}:`, reverse_order_temperatures(t.input) === t.expected || JSON.stringify(reverse_order_temperatures(t.input)) === JSON.stringify(t.expected) ? "PASS" : "FAIL");
  });
}

// done

File: selfstudy/golang/temperatures_reverse.go
/*
task 248:
Temperatures - Reverse
You need to reverse the order of daily temperatures in Celsius
*/

package main

import (
	"fmt"
	"reflect"
)

func Reverse_orderTemperatures(temps []int) []int {
	res := make([]int, len(temps))
	for i, v := range temps {
		res[len(temps)-1-i] = v
	}
	return res
}

func main() {
	tests := []struct {
		input []int
		expected []int
	}{
		{[]int{10, 20, 30}, Reverse_orderTemperatures([]int{10, 20, 30})},
		{[]int{50, 60, 70}, Reverse_orderTemperatures([]int{50, 60, 70})},
		{[]int{1, 2, 3}, Reverse_orderTemperatures([]int{1, 2, 3})},
		{[]int{100, 200}, Reverse_orderTemperatures([]int{100, 200})},
		{[]int{}, Reverse_orderTemperatures([]int{})},
	}
	
	for idx, t := range tests {
		res := Reverse_orderTemperatures(t.input)
		status := "FAIL"
		if reflect.DeepEqual(res, t.expected) {
			status = "PASS"
		}
		fmt.Printf("Test %d: %s\n", idx+1, status)
	}
}

// done
